File: frontend/frontend/src/types/student.ts
export interface StudentInput {
  age: number;
  degree: string;
  branch: string;
  cgpa: number;
  backlogs: number;
  internships: number;
  certifications: number;
  coding_skills: number;
  communication_skills: number;
  aptitude_score: number;
  projects: number;
}

export interface PredictionResponse {
  prediction: number;
  status: string;
  placement_probability: number;
}

File: backend/app/schemas/student.py
from pydantic import BaseModel, Field


class StudentInput(BaseModel):
    age: int = Field(..., ge=18, le=60)
    degree: str
    branch: str
    cgpa: float = Field(..., ge=0, le=10)
    backlogs: int = Field(..., ge=0)
    internships: int = Field(..., ge=0)
    certifications: int = Field(..., ge=0)
    coding_skills: float = Field(..., ge=0, le=10)
    communication_skills: float = Field(..., ge=0, le=10)
    aptitude_score: float = Field(..., ge=0, le=100)
    projects: int = Field(..., ge=0)